src/Interop/src/Hook.cpp

```diff
@@ -175,7 +175,7 @@ namespace OpenLoco::Interop
         if (!_hookTableAddress)
         {
 #ifdef _WIN32
-            _hookTableAddress = VirtualAllocEx(GetCurrentProcess(), NULL, size, MEM_COMMIT | MEM_RESERVE, PAGE_READWRITE);
+            _hookTableAddress = VirtualAllocEx(GetCurrentProcess(), NULL, size, MEM_COMMIT | MEM_RESERVE, PAGE_EXECUTE);
```

Comment:
Are you sure this is correct? You are taking away read and write from this, if you only want this to be execute and read it should be PAGE_EXECUTE_READ

Reply:
Looked at the code, it later wants to write to this memory, but it has no write access anymore.